### PR TITLE
feat: disable some gas warnings and errors for Mantle network

### LINKED ISSUE #23534: Remove the maximum limit constraint on Mantle Mainnet
### What is this about?

For instance, When Mantle's estimated gas is 269580441, altering this value triggers a prompt: "Gas limit must be greater than 20999 and less than 7920027."

Please remove this kind of maximum limit constraint.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [X] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Disable certain gas warnings and errors for Mantle network

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23534

## **Manual testing steps**

1. Switch to mantle network
2. Submit a transaction and change gas limit to very low / high
3. It should not show error

## **Screenshots/Recordings**
Screenshot 2024-05-29 at 2 00 29 PM
## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js [hunk 1]
```diff
@@ -27,17 +30,27 @@ const AdvancedGasFeeGasLimit = () => {
...
   const [gasLimitError, setGasLimitError] = useState();
+  const chainId = useSelector(getCurrentChainId);
 
   const updateGasLimit = (value) => {
     setGasLimit(value);
   };
 
   useEffect(() => {
     setGasLimitInContext(gasLimit);
+    if (chainId === CHAINLIST_CHAIN_IDS_MAP.MANTLE) {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js:41]: Is it worth creating an array so we can add to this in future, such as `IGNORE_GAS_LIMIT_CHAIN_IDS`?

File: ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js [hunk 2]
```diff
@@ -20,14 +24,20 @@ import AdvancedGasFeeInputSubtext from '../../advanced-gas-fee-input-subtext';
...
+) => {
   const baseFeeValue = new Numeric(value, 10);
   if (new Numeric(maxPriorityFeePerGas, 10).greaterThan(baseFeeValue)) {
     return 'editGasMaxBaseFeeGWEIImbalance';
   }
   if (
     gasFeeEstimates?.low &&
-    baseFeeValue.lessThan(gasFeeEstimates.low.suggestedMaxFeePerGas, 10)
+    baseFeeValue.lessThan(gasFeeEstimates.low.suggestedMaxFeePerGas, 10) &&
+    chainId !== CHAINLIST_CHAIN_IDS_MAP.MANTLE
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:40]: Same as the other comment, could we reference an array constant or a util here such as `isChainExemptFromGasLimits`?

File: shared/constants/network.ts [hunk 3]
```diff
@@ -158,7 +158,7 @@ export const CHAIN_IDS = {
   CHZ: '0x15b38',
 } as const;
 
-const CHAINLIST_CHAIN_IDS_MAP = {
+export const CHAINLIST_CHAIN_IDS_MAP = {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply shared/constants/network.ts:162]: I agree, its quite opposite currently, `CHAINLIST_CHAIN_IDS_MAP` includes `CHAIN_IDS`.

File: ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js (hunk 2 above)
Comment (jpuri) [reply ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js]: Done 👍

File: ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js (hunk 1 above)
Comment (jpuri) [reply ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js]: Done 👍

File: ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js [hunk 4]
```diff
@@ -27,17 +30,27 @@ const AdvancedGasFeeGasLimit = () => {
   const [isEditing, setEditing] = useState(false);
   const [gasLimit, setGasLimit] = useState(gasLimitInTransaction);
   const [gasLimitError, setGasLimitError] = useState();
+  const chainId = useSelector(getCurrentChainId);
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js]: Done 👍

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Pull request merged**